msbuild/tests/Xamarin.iOS.Tasks.Tests/ProjectsTests/CodesignAppBundle.cs

```diff
@@ -1,105 +1,147 @@
+
+			args.Add ("--verify");
+			args.AddQuoted (path);
+
+			psi.Arguments = args.ToString ();
+
+			var process = Process.Start (psi);
+			process.WaitForExit ();
+
+			return process.ExitCode == 0;
+		}
+
+		void AssertProperlyCodesigned ()
+		{
+			foreach (var dylib in Directory.EnumerateFiles (AppBundlePath, "*.dylib", SearchOption.AllDirectories))
+				Assert.IsTrue (IsCodesigned (dylib), "{0} is not properly codesigned.", dylib);
 		}
 
-		[Ignore] // requires msbuild instead of xbuild
 		[Test]
 		public void RebuildNoChanges ()
 		{
-			BuildProject ("MyTabbedApplication", Platform, "Release", clean: true);
+			BuildProject ("MyTabbedApplication", Platform, config, clean: true);
+
+			AssertProperlyCodesigned ();
 
-			var dsymDir = Path.GetFullPath (Path.Combine (AppBundlePath, "..", "MyTabbedApplication.app.dSYM"));
-			var appexDsymDir = Path.GetFullPath (Path.Combine (AppBundlePath, "..", "MyActionExtension.appex.dSYM"));
+			//var dsymDir = Path.GetFullPath (Path.Combine (AppBundlePath, "..", "MyTabbedApplication.app.dSYM"));
+			//var appexDsymDir = Path.GetFullPath (Path.Combine (AppBundlePath, "..", "MyActionExtension.appex.dSYM"));
 
-			var timestamps = Directory.EnumerateFiles (AppBundlePath, "*.*", SearchOption.AllDirectories).ToDictionary (file => file, file => GetLastModified (file));
-			var dsymTimestamps = Directory.EnumerateFiles (dsymDir, "*.*", SearchOption.AllDirectories).ToDictionary (file => file, file => GetLastModified (file));
-			var appexDsymTimestamps = Directory.EnumerateFiles (appexDsymDir, "*.*", SearchOption.AllDirectories).ToDictionary (file => file, file => GetLastModified (file));
+			//var timestamps = Directory.EnumerateFiles (AppBundlePath, "*.*", SearchOption.AllDirectories).ToDictionary (file => file, file => GetLastModified (file));
+			//var dsymTimestamps = Directory.EnumerateFiles (dsymDir, "*.*", SearchOption.AllDirectories).ToDictionary (file => file, file => GetLastModified (file));
+			//var appexDsymTimestamps = Directory.EnumerateFiles (appexDsymDir, "*.*", SearchOption.AllDirectories).ToDictionary (file => file, file => GetLastModified (file));
 
 			Thread.Sleep (1000);
 
 			// Rebuild w/ no changes
-			BuildProject ("MyTabbedApplication", Platform, "Release", clean: false);
+			BuildProject ("MyTabbedApplication", Platform, config, clean: false);
+
+			AssertProperlyCodesigned ();
 
-			var newTimestamps = Directory.EnumerateFiles (AppBundlePath, "*.*", SearchOption.AllDirectories).ToDictionary (file => file, file => GetLastModified (file));
-			var newDsymTimestamps = Directory.EnumerateFiles (dsymDir, "*.*", SearchOption.AllDirectories).ToDictionary (file => file, file => GetLastModified (file));
-			var newAppexDsymTimestamps = Directory.EnumerateFiles (appexDsymDir, "*.*", SearchOption.AllDirectories).ToDictionary (file => file, file => GetLastModified (file));
+			// Note: the following checks all require msbuild instead of xbuild to work properly
```

Comment:
Can you check at runtime if MSBuild is used and only execute the asserts in that case?

I'm fairly sure we'll forget to uncomment when we switch to msbuild, so a runtime check would be better...